src/mbgl/geometry/dem_data.cpp

```diff
@@ -3,62 +3,44 @@
 
 namespace mbgl {
 
-DEMData::DEMData(const PremultipliedImage& _image, Tileset::DEMEncoding encoding):
+DEMData::DEMData(const PremultipliedImage& _image, Tileset::DEMEncoding _encoding):
     dim(_image.size.height),
     // extra two pixels per row for border backfilling on either edge
     stride(dim + 2),
+    encoding(_encoding),
     image({ static_cast<uint32_t>(stride), static_cast<uint32_t>(stride) }) {
 
     if (_image.size.height != _image.size.width){
         throw std::runtime_error("raster-dem tiles must be square.");
     }
 
-    auto decodeMapbox = [] (const uint8_t r, const uint8_t g, const uint8_t b){
-        // https://www.mapbox.com/help/access-elevation-data/#mapbox-terrain-rgb
-        return (r * 256 * 256 + g * 256 + b)/10 - 10000;
-    };
-
-    auto decodeTerrarium = [] (const uint8_t r, const uint8_t g, const uint8_t b){
-        // https://aws.amazon.com/public-datasets/terrain/
-        return ((r * 256 + g + b / 256) - 32768);
-    };
-
-    auto decodeRGB = encoding == Tileset::DEMEncoding::Terrarium ? decodeTerrarium : decodeMapbox;
-
-    std::memset(image.data.get(), 0, image.bytes());
-
+    auto dest = reinterpret_cast<uint32_t*>(image.data.get()) + stride + 1;
+    auto source = reinterpret_cast<uint32_t*>(_image.data.get());
     for (int32_t y = 0; y < dim; y++) {
-        for (int32_t x = 0; x < dim; x++) {
-            const int32_t i = y * dim + x;
-            const int32_t j = i * 4;
-            set(x, y, decodeRGB(_image.data[j], _image.data[j+1], _image.data[j+2]));
-        }
+        memcpy(dest, source, dim * 4);
```

Comment:
would it create 2px padding on the right side of a stride?

Reply:
No, L17: dest is (stride + 1) initialized to item at (1, 1)

Reply:
ah, missed that. tnx